manager/controlapi/node.go

```diff
@@ -208,6 +212,19 @@ func (s *Server) UpdateNode(ctx context.Context, request *api.UpdateNodeRequest)
 			}
 		}
 
+		member := s.raft.Cluster.GetMemberByNodeID(request.NodeID)
+		raftID = member.RaftID
+
+		if demote && s.raft != nil {
+			if !s.raft.Cluster.CanRemoveMember(s.raft.Config.ID, raftID) {
+				return grpc.Errorf(codes.FailedPrecondition, "attempting to demote a manager while this might result in a loss of quorum")
+			}
+
+			if err := s.raft.RemoveMember(ctx, raftID); err != nil {
```

Comment:
Why is this inside an Update callback? It will hold the memory store lock for potentially quite a long time.